lib/cocoapods-spm/def/spm_package.rb

```diff
@@ -55,6 +55,10 @@ def use_default_xcode_linking?
         @linking_opts.fetch(:use_default_xcode_linking, false)
       end
 
+      def available_in_top_level_target?
```

Comment:
A package can be used in different main (root level) targets. So I think in the future, we may need to opt out for some of the main targets only. Consider the following example:
```rb
spm_package "A", ...

target "X" do
  ...
end

target "Y" do
  ...
end
```
By default, A is linked to both X and Y. And we may wish to link A to X, but not Y, or reversed.

Im wondering if we should modify the option to let user specify the targets we exclude the linking, as follows:


```
spm_pkg "A", 
        :linking => {"exclude" => ["App"]} # <--- HERE
```

Reply:
You're right. I copy pasted the `spm_pkg "A"` part for every target to deal with this, but your solution might be more convenient.

On another note, I'm not sure if this will work anyway. After I started using B in A I got new symbol not found errors, even when B has `private import MQTTNIO`. So I need to figure that out first (and now my Pods project is getting corrupt again too). Total chaos on my side for now, but I'll get back to this. 🥲